nemo/collections/nlp/data/language_modeling/megatron/gpt_dataset.py

```diff
@@ -208,8 +237,59 @@ def __getitem__(self, idx):
             # And finally add the relevant portion of last document.
             sample_list.append(self.indexed_dataset.get(self.doc_idx[doc_index_l], length=offset_l + 1))
             sample = np.concatenate(sample_list)
+        return sample.astype(np.int64)
 
-        return {'text': np.array(sample, dtype=np.int64)}
+    def __getitem__(self, idx):
+        text = torch.from_numpy(self._get_text(idx))
+        tokens = text[:-1].contiguous()
+        labels = text[1:].contiguous()
+        attention_mask, loss_mask, position_ids = _create_ltor_masks_and_position_ids(
+            tokens, self.eos_id, self.reset_position_ids, self.reset_attention_mask, self.eod_mask_loss,
+        )
+
+        return {
+            'tokens': tokens,
+            'labels': labels,
+            'attention_mask': attention_mask,
+            'loss_mask': loss_mask,
+            'position_ids': position_ids,
+        }
+
+
+@torch.no_grad()
+def _create_ltor_masks_and_position_ids(
+    tokens: torch.Tensor, eod_token: int, reset_position_ids: bool, reset_attention_mask: bool, eod_mask_loss: bool,
+):
+    """Modified version of get_ltor_masks_and_position_ids in nemo/collections/nlp/modules/common/megatron/utils.py"""  # NOQA
```

Comment:
Can you make the comment a little more clear on how/why it was modified for micro vs global batch.